repl/src/main/scala/ammonite/repl/interp/Evaluator.scala

```diff
@@ -219,6 +296,15 @@ object Evaluator{
   def evaluatorRunPrinter(f: => Unit) = f
 
   /**
+   * This gives our cache tags for compile caching.
+   */
+  def cacheTag(code: String, imports: Seq[ImportData], classpath: String): String = {
```

Comment:
A few points here:

- We could pass in a `Seq[String]` here instead of `String`, since that's what you had outside the function.
- We most likely really want a `Seq[Array[Byte]]` because otherwise successive identically-named `-SNAPSHOT` jars will confuse this
- If we're hashing the bytes of each jar, we might want to cache the hashed values somewhere because it could get expensive
- If we're combining multiple hashes, we probably want to do it the "proper" way i.e. http://stackoverflow.com/a/4785309/871202; if it turns out to be slow we could use a weaker hash, but we shouldn't do ad-hoc string to combine the keys